lib/hypervisor/hv_kvm/__init__.py

```diff
@@ -2467,7 +2453,25 @@ def _StopInstance(self, instance, force=False, name=None, timeout=None):
       else:
         self.qmp.Powerdown()
         if timeout is not None:
-          self._WaitForInstanceToShutdown(timeout, name)
+          tick = 1
+          while tick <= timeout:
+            _, _, alive = self._InstancePidAlive(name)
+            if not alive:
+              logging.info("KVM: instance %s finished shutdown after %d seconds"
+                           % (name, tick))
+              break
+
+            # re-issue ACPI shutdown events every 10 seconds
+            # this works around instances which did not react to earlier events
+            # because they were e.g. still booting
+            # this _mostly_ addresses the QA suite which starts and stops
+            # instances in short order, sometimes triggering 120 second
+            # default timeout
+            if not tick % 10:
```

Comment:
This is a bit of a double negation and needs some brain cycles to parse correctly. I would write it explicitly as
```
if tick % 10 == 0:
```

Reply:
You are absolutely right here. However, this part of the code is now obsolete :-)